2002-02-03  H.J. Lu <hjl@gnu.org>

	* libtool.m4 (lt_cv_deplibs_check_method): Updated for
	Linux/mips.

2001-10-22  H.J. Lu <hjl@gnu.org>

	* ltmain.sh: Allow link against an archive when building a
	shared library.

	* libtool.m4 (lt_cv_deplibs_check_method): Always use
	file_magic for Linux ELF.

Index: libtool.m4
===================================================================
RCS file: /export//cvs/gnu/binutils/libtool.m4,v
retrieving revision 1.1.1.9
retrieving revision 1.7
diff -u -p -r1.1.1.9 -r1.7
--- libtool.m4	27 Feb 2003 18:48:53 -0000	1.1.1.9
+++ libtool.m4	27 Feb 2003 19:27:35 -0000	1.7
@@ -635,13 +635,7 @@ irix5* | irix6*)
 
 # This must be Linux ELF.
 linux-gnu*)
-  case $host_cpu in
-  alpha* | hppa* | i*86 | powerpc* | sparc* | ia64* )
-    lt_cv_deplibs_check_method=pass_all ;;
-  *)
-    # glibc up to 2.1.1 does not perform some relocations on ARM
-    lt_cv_deplibs_check_method=['file_magic ELF [0-9][0-9]*-bit [LM]SB (shared object|dynamic lib )'] ;;
-  esac
+  lt_cv_deplibs_check_method=['file_magic ELF [0-9][0-9]*-bit [LM]SB( .*|) (shared object|dynamic lib )']
   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
   ;;
 
